### PR TITLE
M3-1400 Refactor services/volumes

### PR DESCRIPTION
**Pull request by Jskobos**
Add comments and schema. This one was simple and didn't break anything.

### CODE REVIEW

File: src/services/volumes/volumes.ts [hunk 1]
```diff
@@ -38,36 +64,92 @@ export const attachVolume = (volumeId: number, payload: {
   setData(payload),
 );
```
Comment (johnwcallahan) [inline src/services/volumes/volumes.ts]: If this request returns a volume, we should resolve the data object (i.e. `then(response => response.data)`) and adjust the components accordingly.

File: src/services/volumes/volumes.schema.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { number, object, string } from 'yup';
+
+export const CreateVolumeSchema = object({
+  region: string()
+    .when('linode_id', {
+      is: (id) => Boolean(id),
+      then: string().required("Must provide a region or a Linode ID."),
+    }),
+  linode_id: number(),
+  size: number(),
```
Comment (martinmckenna) [inline src/services/volumes/volumes.schema.ts]: prob should add a message here. The default one is a bit weird

File: src/services/volumes/volumes.schema.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { number, object, string } from 'yup';
+
+export const CreateVolumeSchema = object({
+  region: string()
+    .when('linode_id', {
+      is: (id) => Boolean(id),
+      then: string().required("Must provide a region or a Linode ID."),
+    }),
+  linode_id: number(),
+  size: number().typeError("Size must be a number.").integer(),
```
Comment (johnwcallahan) [inline src/services/volumes/volumes.schema.ts]: One thing I didn't catch before: we should set `min` and `max` here. It's not in the documentation, but the API will return an error: 
screen shot 2018-10-29 at 9 54 26 am

Comment (nwisnewski) [reply src/services/volumes/volumes.schema.ts]: I have verified the changes requested, volume size min and max enforced without sending the request.

**Pull request merged**